extensions/wikia/PortabilityDashboard/SpecialPortabilityDashboardController.php

```diff
@@ -14,53 +15,54 @@ public function __construct() {
 	}
 
 	public function index() {
-		$model = new PortabilityDashboardModel();
-		$list = $model->getList();
-		$langFilter = $this->getVal( self::LANGUAGE_FILTER_QS_PARAM, '' );
+		$langFilter = $this->getVal( static::LANGUAGE_FILTER_QS_PARAM, '' );
+		$wikiUrlParam = $this->getVal( static::WIKI_URL_QS_PARAM, '' );
 		$isLangFilterSet = !empty( $langFilter );
+		$list = $this->getResultsList( $langFilter, $wikiUrlParam );
+		$noResultsInfoMessage = wfMessage(
+			$isLangFilterSet ? 'portability-dashboard-no-results-for-lang-info' : 'portability-dashboard-no-results-for-url-info',
+			PortabilityDashboardModel::WIKIS_LIMIT
+		)->text();
 
-		// template model 
-		$this->response->setVal( 'list', $isLangFilterSet ? $this->filterListByLang( $list, $langFilter ) : $list );
+		$this->response->setTemplateEngine( WikiaResponse::TEMPLATE_ENGINE_MUSTACHE );
+		$this->response->setValues( [
+			// template model
+			'list' => $list,
+			// template helpers
+			'typelessTemplatesInsightsPath' =>  self::SPECIAL_INSIGHTS_PATH . self::SPECIAL_INSIGHTS_TYPELESS_TEMPLATE_PAGE,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)